### PR TITLE
build: make boto and requests optional dependencies

### PR DESCRIPTION
**Pull request by alexeykuzmin**
##### Description of Change

We normally don't need both `boto` and `requests`.
Let's checkout them only when we are about to publish a release.

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: no notes

### CODE REVIEW

File: vsts.yml [hunk 1]
```diff
@@ -17,6 +17,8 @@ jobs:
       gclient config \
         --name "src/electron" \
         --unmanaged \
+        --custom-var=checkout_boto=True \
```
Comment (jkleinsc) [inline vsts.yml]: This too can be conditional using the condition syntax, eg:
```condition: and(succeeded(), eq(variables['ELECTRON_RELEASE'], '1'))```

**Comment by alexeykuzmin** [comment]
> update Appveyor and VSTS to conditionally include gclient config args depending on whether or not we are running a release build.

@jkleinsc Done.

File: vsts.yml [hunk 2]
```diff
@@ -13,10 +13,15 @@ jobs:
       export PATH="$PATH:/Users/electron/depot_tools"
       echo "##vso[task.setvariable variable=PATH]$PATH"
       export GIT_CACHE_PATH="/Users/electron/libcc_cache"
+      if [ "ELECTRON_RELEASE" != "" ]; then
```
Comment (jkleinsc) [inline vsts.yml]: Should be `if [ "$ELECTRON_RELEASE" == "1" ]; then`

Comment (alexeykuzmin) [reply vsts.yml]: Is there a different between an empty var check and exact value of `1`  check?
Oh, I even forgot the `$` sign )

Comment (alexeykuzmin) [reply vsts.yml]: Fixed.

**Comment by alexeykuzmin** [comment]
@jkleinsc Does it looks now?
**electron-mas-testing** is irrelevant.

**Pull request merged**